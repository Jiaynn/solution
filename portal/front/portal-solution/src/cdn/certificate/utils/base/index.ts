/*
 * @file base utils
 * @author nighca <nighca@live.cn>
 */

import moment from 'moment'

import { RouterStore } from 'portal-base/common/router'

export interface IPayOrderOptions {
  orderId: string       // fusion 订单号
  tradeOrderId: string  // BO 订单号
}

export function payForOrder({ tradeOrderId }: IPayOrderOptions, routerStore: RouterStore) {
  routerStore.push(`/financial/verify-order/${tradeOrderId}`)
}

export function humanizeTime(time: number, format?: string) {
  return format
    ? moment(time * 1000).format(format)
    : moment(time * 1000).format('YYYY-MM-DD HH:mm:ss')
}
