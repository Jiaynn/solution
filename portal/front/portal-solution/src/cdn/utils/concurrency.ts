/**
 * @file concurrency limit request
 * @source https://gist.github.com/nighca/6562d098ac01814b6e1c1718b16d4abc
 */

export default function concurrency<T>(process: (_: any) => Promise<T>, limit = -1) {
  return function batchProcess(tasks: unknown[] = []): Promise<T[]> {
    const results: T[] = []
    let finished = 0
    let current = 0
    let rejected = false

    function tryProcess(resolve: (...args: any) => any, reject: (...args: any) => any) {
      if (rejected) return
      if (finished >= tasks.length) return resolve(results)
      if (current >= tasks.length) return

      const index = current++
      process(tasks[index]).then(
        result => {
          results[index] = result
          finished++
          tryProcess(resolve, reject)
        },
        err => {
          reject(err)
          rejected = true
        }
      )
    }

    return new Promise((resolve, reject) => {
      const realLimit = limit > 0 ? limit : tasks.length
      for (let i = 0; i < realLimit; i++) {
        tryProcess(resolve, reject)
      }
    })
  }
}
