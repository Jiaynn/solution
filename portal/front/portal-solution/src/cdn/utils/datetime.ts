import moment from 'moment'

export function isLastDayOfMonth() {
  return moment().endOf('month').format('MM-DD') === moment().format('MM-DD')
}
