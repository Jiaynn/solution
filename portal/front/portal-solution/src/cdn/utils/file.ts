export function saveFile(filename: string, content: string) {
  const blob = new Blob([content])
  const a = document.createElement('a')
  const url = window.URL.createObjectURL(blob)
  a.href = url
  a.download = filename
  a.click()
  window.URL.revokeObjectURL(url)
}
