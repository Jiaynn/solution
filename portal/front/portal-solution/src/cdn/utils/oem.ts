import { InternalOptions, withHeader } from 'qn-fe-core/client'

import { isOEM, oemVendor } from 'cdn/constants/env'

export function withOemVendor(options: InternalOptions) {
  if (isOEM) {
    return {
      ...options,
      headers: withHeader(options.headers, 'X-Fusion-Vendor', String(oemVendor), true)
    }
  }

  return options
}
