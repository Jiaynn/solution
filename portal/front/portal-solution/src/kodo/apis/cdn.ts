/**
 * @file config api
 * @description region 相关的接口
 * @author lizhifeng <lizhifeng@qiniu.com>
 */

import autobind from 'autobind-decorator'
import { injectable } from 'qn-fe-core/di'
import { ApiException } from 'qn-fe-core/client'

import { ProxyClientV2 } from 'portal-base/common/apis/proxy'

import { proxy } from 'kodo/constants/apis'

interface CdnBaseResponse {
  code: number
  error: string
  [k: string]: any
}

export interface RefreshCdnSurplus extends CdnBaseResponse {
  urlQuotaDay: number // url 刷新当日限额
  urlSurplusDay: number // url 刷新当日余量
  dirQuotaDay: number // 目录刷新当日限额
  dirSurplusDay: number // 目录刷新当日余量
  quotaDay: number // 预取当日限额
  surplusDay: number // 预取当日剩余量
}

interface RefreshCdnOptions {
  product: 'cdn'
  urls: string[]
}

export class CdnApiException extends ApiException {
  constructor(
    public code: number,
    public payload: CdnBaseResponse,
    message?: string,
    cause?: unknown
  ) {
    super(
      'RefreshPrefetchApiException',
      code,
      message,
      undefined,
      cause
    )
  }
}

const messageMap = new Map<number, string>([
  [400031, '无效的 URL'],
  [400034, '刷新域名个数超过限制'],
  [400037, 'URL 已在队列中，无需重复提交'],
  [400039, '请求次数过多']
])

function produceResult<T extends CdnBaseResponse>(payload: T) {
  if (payload && typeof payload === 'object' && 'code' in payload && payload.code !== 200) {
    const message = messageMap.get(payload.code) || payload.error
    throw new CdnApiException(payload.code, payload, message)
  }

  return payload
}

@autobind
@injectable()
export class CdnApis {
  constructor(
    private proxyClient: ProxyClientV2
  ) {}

  getRefreshCdnSurplus(): Promise<RefreshCdnSurplus> {
    return this.proxyClient.get<RefreshCdnSurplus>(proxy.refreshCdnSurplus, {})
      .then(produceResult)
  }

  async refreshCdn(params: RefreshCdnOptions): Promise<void> {
    await this.proxyClient.post<CdnBaseResponse>(proxy.refreshCdn, params)
      .then(produceResult)
  }
}
