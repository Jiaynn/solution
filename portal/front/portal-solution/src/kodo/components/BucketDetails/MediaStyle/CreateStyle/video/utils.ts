export enum SourceFormat {
  Mp4 = 'mp4',
  Flv = 'flv',
  Mov = 'mov',
}

export enum OutputFormat {
  Jpg = 'jpg',
  Png = 'png',
  Gif = 'gif'
}

export enum CoverType {
  Static = 'static',
  Dynamic = 'dynamic'
}

export const coverTypeNameMap = {
  [CoverType.Static]: '静态封面',
  [CoverType.Dynamic]: '动态封面'
}

export const coverTypes = [CoverType.Static] // [CoverType.Static, CoverType.Dynamic]
export const sourceFormatList = [SourceFormat.Mp4, SourceFormat.Flv, SourceFormat.Mov]
export const allowPreviewMimeTypeList = [
  'video/mp4',
  'video/x-flv',
  'video/quicktime'
]

export const allowPickAccept = {
  name: '视频',
  maxSize: 500 * 1024 * 1024,
  mimeTypes: allowPreviewMimeTypeList
}

export const outputFormatList = [OutputFormat.Jpg, OutputFormat.Png, OutputFormat.Gif]
export const staticCoverSuffixes = [OutputFormat.Jpg, OutputFormat.Png]
export const dynamicCoverSuffixes = [OutputFormat.Gif]

export enum CoverScaleType {
  None = 'none',
  // 限定宽, 高同比例缩放
  ScaleWidth = 'scaleWidth',
  // 限定高, 宽同比例缩放
  ScaleHeight = 'scaleHeight',
  // 限定宽高同比例缩放
  ScaleWidthHeight = 'scaleWidthHeight',
}

export const coverScaleTypes = [
  CoverScaleType.None,
  CoverScaleType.ScaleWidth,
  CoverScaleType.ScaleHeight,
  CoverScaleType.ScaleWidthHeight
]

export const coverScaleTypeNameMap = {
  [CoverScaleType.None]: '不缩放',
  [CoverScaleType.ScaleWidth]: '限定宽、高同比例缩放',
  [CoverScaleType.ScaleHeight]: '限定高、宽同比例缩放',
  [CoverScaleType.ScaleWidthHeight]: '限定宽高同比例缩放'
}

export enum CoverAutoScaleType {
  None = 0,
  One = 1,
  Two = 2
}

export const coverAutoScaleTypeNameMap = {
  [CoverAutoScaleType.None]: '不启用',
  [CoverAutoScaleType.One]: 1,
  [CoverAutoScaleType.Two]: 2
}

export const coverAutoScaleTypes = [CoverAutoScaleType.None, CoverAutoScaleType.One, CoverAutoScaleType.Two]

export function getCoverScaleType(w: number | null, h: number | null) {
  if (w == null && h == null) {
    return CoverScaleType.None
  }
  if (w != null && h == null) {
    return CoverScaleType.ScaleWidth
  }

  if (w == null && h != null) {
    return CoverScaleType.ScaleHeight
  }

  return CoverScaleType.ScaleWidthHeight
}

export function isValidSourceFormat(format: string): format is SourceFormat {
  return sourceFormatList.includes(format as any)
}

export function encodeKodoURI(bucketName: string, fileKey: string) {
  return `kodo://${bucketName}/${fileKey}`
}

export function decodeKodoURI(URI: string): [bucketName: string, fileKey: string] {
  const execArray = /kodo:\/\/([^/]+)\/(.*)/.exec(URI)
  if (execArray) {
    return execArray.slice(1, 3) as [string, string]
  }
  // 不符合规则时，直接把输入当作 fileKey 返回
  return ['', URI]
}
