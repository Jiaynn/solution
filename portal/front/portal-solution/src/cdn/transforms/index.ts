import { QueryParams } from 'qn-fe-core/utils'

export function atoi(numberText: string, valueForInvalidInput = 0): number {
  const parsed = parseInt(numberText, 10)
  return Number.isNaN(parsed) ? valueForInvalidInput : parsed
}

export function atof(numberText: string, valueForInvalidInput = 0): number {
  const parsed = parseFloat(numberText)
  return Number.isNaN(parsed) ? valueForInvalidInput : parsed
}

export function getIntValue(e: React.FormEvent<any>, valueForInvalidInput = 0) {
  return atoi((e.target as any).value, valueForInvalidInput)
}

export function getFloatValue(e: React.FormEvent<any>, valueForInvalidInput = 0) {
  return atof((e.target as any).value, valueForInvalidInput)
}

export function trimAndFilter(lines: string[]): string[] {
  return (lines || []).map(
    line => line.trim()
  ).filter(
    line => !!line
  )
}

export function splitLines(linesStr: string | null | undefined): string[] {
  if (!linesStr) {
    return []
  }
  return linesStr.split('\n')
}

export function filterQuery(query?: QueryParams) {
  if (!query) {
    return
  }
  return Object.keys(query).reduce(
    (filtered, key) => (
      (query || {})[key]
      ? { ...filtered, [key]: (query || {})[key] }
      : filtered
    ), {} as QueryParams
  )
}

export function humanizeOnoff(isOn: boolean) {
  return isOn ? '开启' : '关闭'
}
