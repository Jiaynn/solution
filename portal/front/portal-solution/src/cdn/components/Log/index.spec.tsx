import React, { PropsWithChildren } from "react";

import { RendererUtils as Renderer } from "cdn/test";

import { DateMocker } from "cdn/test/utils";

jest.mock("react-icecream/lib/spin", () => ({
  __esModule: true,
  default: (props: PropsWithChildren<{}>) => <div>{props.children}</div>,
}));

import LogManage from "./index";

it("renders correctly", () =>
  new DateMocker().mock(() => {
    const tree = new Renderer().createWithAct(<LogManage />).toJSON();
    expect(tree).toMatchSnapshot();
  }));
