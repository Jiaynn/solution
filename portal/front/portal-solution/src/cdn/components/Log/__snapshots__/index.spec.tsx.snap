// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="comp-log-manage-wrapper"
>
  <div
    className="icecream-breadcrumb"
  />
  <div
    className=""
  >
    <div
      className="notes"
    >
      <p>
        日志默认保留时间为 30 天，生成大概延迟 6 小时；
      </p>
      <p>
        如果日志文件过大，您可以参见
        <a
          href="https://developer.qiniu.com/article/fusion/api/log.html"
          rel="noopener"
          target="_blank"
        >
           API 文档下载日志
        </a>
        ；
      </p>
      <p>
        关于日志格式请参考
        <a
          href="https://support.qiniu.com/hc/kb/article/160890/"
          rel="noopener"
          target="_blank"
        >
           CDN日志格式简介 
        </a>
        。
      </p>
    </div>
    <div>
      <form
        className="ant-form ant-form-inline log-filter-wrapper"
        onSubmit={[Function]}
      >
        <div
          className="ant-row ant-form-item"
        >
          <div
            className="ant-col ant-form-item-label"
          >
            <label
              className="ant-form-item-no-colon"
              htmlFor={null}
              onClick={[Function]}
              title="选择日期"
            >
              选择日期
            </label>
          </div>
          <div
            className="ant-col ant-form-item-control-wrapper"
          >
            <div
              className="ant-form-item-control"
            >
              <span
                className="ant-form-item-children"
              >
                <span
                  className="ant-calendar-picker"
                  onBlur={[Function]}
                  onFocus={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  style={
                    Object {
                      "marginLeft": "8px",
                      "width": "240px",
                    }
                  }
                  tabIndex={0}
                >
                  <span
                    className="ant-calendar-picker-input ant-input"
                    onClick={[Function]}
                    onKeyDown={[Function]}
                    onMouseDown={[Function]}
                    onTouchStart={[Function]}
                  >
                    <input
                      className="ant-calendar-range-picker-input"
                      placeholder="Start date"
                      readOnly={true}
                      tabIndex={-1}
                      value="2018-02-01"
                    />
                    <span
                      className="ant-calendar-range-picker-separator"
                    >
                       
                      ~
                       
                    </span>
                    <input
                      className="ant-calendar-range-picker-input"
                      placeholder="End date"
                      readOnly={true}
                      tabIndex={-1}
                      value="2018-02-01"
                    />
                    <i
                      aria-label="icon: calendar"
                      className="anticon anticon-calendar ant-calendar-picker-icon"
                    >
                      <svg
                        aria-hidden="true"
                        className=""
                        data-icon="calendar"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
                        />
                      </svg>
                    </i>
                  </span>
                </span>
              </span>
            </div>
          </div>
        </div>
        <div
          className="ant-row ant-form-item"
        >
          <div
            className="ant-col ant-form-item-label"
          >
            <label
              className="ant-form-item-no-colon"
              htmlFor={null}
              onClick={[Function]}
              title="选择域名"
            >
              选择域名
            </label>
          </div>
          <div
            className="ant-col ant-form-item-control-wrapper"
          >
            <div
              className="ant-form-item-control"
            >
              <span
                className="ant-form-item-children"
              >
                <div
                  style={
                    Object {
                      "marginLeft": "8px",
                    }
                  }
                >
                  <div
                    className="comp-domain-selector"
                  >
                    <span
                      className="ant-input-group ant-input-group-compact"
                    >
                      <div
                        className="comp-domain-selector"
                      >
                        <div
                          onClick={[Function]}
                          onMouseDown={[Function]}
                          onTouchStart={[Function]}
                        >
                          <div
                            className="ant-select ant-select-enabled ant-select-loading"
                            onBlur={[Function]}
                            onClick={[Function]}
                            onFocus={[Function]}
                            onMouseDown={[Function]}
                            onMouseOut={[Function]}
                            onMouseUp={[Function]}
                            onTouchStart={[Function]}
                            style={
                              Object {
                                "width": 200,
                              }
                            }
                          >
                            <div
                              aria-autocomplete="list"
                              aria-controls="test-uuid"
                              aria-expanded={false}
                              aria-haspopup="true"
                              className="ant-select-selection
            ant-select-selection--single"
                              onKeyDown={[Function]}
                              role="combobox"
                              tabIndex={0}
                            >
                              <div
                                className="ant-select-selection__rendered"
                              >
                                <div
                                  className="ant-select-selection-selected-value"
                                  style={
                                    Object {
                                      "display": "block",
                                      "opacity": 1,
                                    }
                                  }
                                  title="已选择全量域名"
                                >
                                  已选择全量域名
                                </div>
                              </div>
                              <span
                                className="ant-select-arrow"
                                onClick={[Function]}
                                style={
                                  Object {
                                    "WebkitUserSelect": "none",
                                    "userSelect": "none",
                                  }
                                }
                                unselectable="on"
                              >
                                <i
                                  aria-label="icon: loading"
                                  className="anticon anticon-loading"
                                >
                                  <svg
                                    aria-hidden="true"
                                    className="anticon-spin"
                                    data-icon="loading"
                                    fill="currentColor"
                                    focusable="false"
                                    height="1em"
                                    viewBox="0 0 1024 1024"
                                    width="1em"
                                  >
                                    <path
                                      d="M988 548c-19.9 0-36-16.1-36-36 0-59.4-11.6-117-34.6-171.3a440.45 440.45 0 0 0-94.3-139.9 437.71 437.71 0 0 0-139.9-94.3C629 83.6 571.4 72 512 72c-19.9 0-36-16.1-36-36s16.1-36 36-36c69.1 0 136.2 13.5 199.3 40.3C772.3 66 827 103 874 150c47 47 83.9 101.8 109.7 162.7 26.7 63.1 40.2 130.2 40.2 199.3.1 19.9-16 36-35.9 36z"
                                    />
                                  </svg>
                                </i>
                              </span>
                            </div>
                          </div>
                        </div>
                      </div>
                    </span>
                  </div>
                </div>
              </span>
            </div>
          </div>
        </div>
        <div
          className="ant-row ant-form-item"
        >
          <div
            className="ant-col ant-form-item-control-wrapper"
          >
            <div
              className="ant-form-item-control"
            >
              <span
                className="ant-form-item-children"
              >
                <button
                  className="ant-btn ant-btn-primary"
                  onClick={[Function]}
                  style={
                    Object {
                      "marginLeft": "24px",
                    }
                  }
                  type="button"
                >
                  <span>
                    搜索
                  </span>
                </button>
              </span>
            </div>
          </div>
        </div>
      </form>
    </div>
    <div
      className="ant-table-wrapper log-list"
    >
      <div
        className="ant-spin-nested-loading"
      >
        <div
          className="ant-spin-container"
        >
          <div
            className="ant-table ant-table-default ant-table-empty ant-table-scroll-position-left"
            style={Object {}}
          >
            <div
              className="ant-table-content"
            >
              <div
                className="ant-table-body"
                onScroll={[Function]}
                onWheel={[Function]}
                style={Object {}}
              >
                <table
                  className=""
                  style={Object {}}
                >
                  <colgroup>
                    <col
                      style={
                        Object {
                          "minWidth": undefined,
                          "width": undefined,
                        }
                      }
                    />
                    <col
                      style={
                        Object {
                          "minWidth": undefined,
                          "width": undefined,
                        }
                      }
                    />
                    <col
                      style={
                        Object {
                          "minWidth": undefined,
                          "width": undefined,
                        }
                      }
                    />
                    <col
                      style={
                        Object {
                          "minWidth": undefined,
                          "width": undefined,
                        }
                      }
                    />
                    <col
                      style={
                        Object {
                          "minWidth": undefined,
                          "width": undefined,
                        }
                      }
                    />
                  </colgroup>
                  <thead
                    className="ant-table-thead"
                  >
                    <tr
                      style={
                        Object {
                          "height": null,
                        }
                      }
                    >
                      <th
                        className=""
                      >
                        <span
                          className="ant-table-header-column"
                        >
                          <div>
                            <span
                              className="ant-table-column-title"
                            >
                              域名任务
                            </span>
                            <span
                              className="ant-table-column-sorter"
                            />
                          </div>
                        </span>
                      </th>
                      <th
                        className=""
                      >
                        <span
                          className="ant-table-header-column"
                        >
                          <div>
                            <span
                              className="ant-table-column-title"
                            >
                              日志时间
                            </span>
                            <span
                              className="ant-table-column-sorter"
                            />
                          </div>
                        </span>
                      </th>
                      <th
                        className=""
                      >
                        <span
                          className="ant-table-header-column"
                        >
                          <div>
                            <span
                              className="ant-table-column-title"
                            >
                              文件大小
                            </span>
                            <span
                              className="ant-table-column-sorter"
                            />
                          </div>
                        </span>
                      </th>
                      <th
                        className=""
                      >
                        <span
                          className="ant-table-header-column"
                        >
                          <div>
                            <span
                              className="ant-table-column-title"
                            >
                              状态
                            </span>
                            <span
                              className="ant-table-column-sorter"
                            />
                          </div>
                        </span>
                      </th>
                      <th
                        className="log-opreate"
                      >
                        <span
                          className="ant-table-header-column"
                        >
                          <div>
                            <span
                              className="ant-table-column-title"
                            >
                              下载日志
                            </span>
                            <span
                              className="ant-table-column-sorter"
                            />
                          </div>
                        </span>
                      </th>
                    </tr>
                  </thead>
                  <tbody
                    className="ant-table-tbody"
                  />
                </table>
              </div>
              <div
                className="ant-table-placeholder"
              >
                <div
                  className="ant-empty ant-empty-normal"
                >
                  <div
                    className="ant-empty-image"
                  >
                    <img
                      alt="No Data"
                      src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNjQiIGhlaWdodD0iNDEiIHZpZXdCb3g9IjAgMCA2NCA0MSIgIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyI+CiAgPGcgdHJhbnNmb3JtPSJ0cmFuc2xhdGUoMCAxKSIgZmlsbD0ibm9uZSIgZmlsbC1ydWxlPSJldmVub2RkIj4KICAgIDxlbGxpcHNlIGZpbGw9IiNGNUY1RjUiIGN4PSIzMiIgY3k9IjMzIiByeD0iMzIiIHJ5PSI3Ii8+CiAgICA8ZyBmaWxsLXJ1bGU9Im5vbnplcm8iIHN0cm9rZT0iI0Q5RDlEOSI+CiAgICAgIDxwYXRoIGQ9Ik01NSAxMi43Nkw0NC44NTQgMS4yNThDNDQuMzY3LjQ3NCA0My42NTYgMCA0Mi45MDcgMEgyMS4wOTNjLS43NDkgMC0xLjQ2LjQ3NC0xLjk0NyAxLjI1N0w5IDEyLjc2MVYyMmg0NnYtOS4yNHoiLz4KICAgICAgPHBhdGggZD0iTTQxLjYxMyAxNS45MzFjMC0xLjYwNS45OTQtMi45MyAyLjIyNy0yLjkzMUg1NXYxOC4xMzdDNTUgMzMuMjYgNTMuNjggMzUgNTIuMDUgMzVoLTQwLjFDMTAuMzIgMzUgOSAzMy4yNTkgOSAzMS4xMzdWMTNoMTEuMTZjMS4yMzMgMCAyLjIyNyAxLjMyMyAyLjIyNyAyLjkyOHYuMDIyYzAgMS42MDUgMS4wMDUgMi45MDEgMi4yMzcgMi45MDFoMTQuNzUyYzEuMjMyIDAgMi4yMzctMS4zMDggMi4yMzctMi45MTN2LS4wMDd6IiBmaWxsPSIjRkFGQUZBIi8+CiAgICA8L2c+CiAgPC9nPgo8L3N2Zz4K"
                    />
                  </div>
                  <p
                    className="ant-empty-description"
                  >
                    No Data
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
